workflow/scripts/write-fp-fn-vcf.py

```
@@ -0,0 +1,81 @@
+def write_vcf(vcf_in: VariantFile, records: list, output_file: str):
+    """Write collected records to a VCF file."""
+    if not records:
+        print("No records to write.", file=sys.stderr)
+        return
+    with VariantFile(output_file, 'w', header=vcf_in.header) as vcf_out:
+        for record in records:
+            vcf_out.write(record)
+    print(f"VCF written to {output_file}", file=sys.stderr)
+
+
+if snakemake.params.get("output") == "shared-fn" or snakemake.params.get("output") == "unique-fn":
```

Ruff (0.12.2)

Undefined name `snakemake`

(F821)